packages/web-pkg/src/ui/types.ts

```diff
@@ -1,10 +1,14 @@
 import { IconFillType } from 'design-system/src/helpers'
+import { defineComponent } from 'vue'
 
-export type ViewMode = {
+export type FolderView = {
   name: string
   label: string
   icon: {
     name: string
     fillType: IconFillType
   }
+  isScrollable?: boolean
+  component: ReturnType<typeof defineComponent>
```

Comment:
This will basically type `component` as `any`. I still haven't figured out component typing in Vue, documentation is a bit lacklustre (or I simply didn't find the right place 😄 ). But I guess you could just use `Component` as type here. It also lets you define required props if you have some (`resources` maybe?).